### PR TITLE
Crud GraphQL Fixes

### PR DESCRIPTION
**Pull request by sharadregoti**
1) Added automatic connection retries for mongo & sql databases.
2) Removed some unused code in utils package
3) Removed the constraint that primary key can be only of type ID or Integer

### CODE REVIEW

File: gateway/modules/crud/mgo/mongo.go [hunk 1]
```diff
@@ -33,13 +36,36 @@ func Init(enabled bool, connection, dbName string, driverConf config.DriverConfi
...
 	return
 }
 
 // Close gracefully the Mongo client
 func (m *Mongo) Close() error {
-	if m.client != nil {
-		return m.client.Disconnect(context.TODO())
+	if m.getClient() != nil {
+		m.connRetryCloserChan <- struct{}{}
+		m.setClient(nil)
```
Comment (YourTechBud) [inline gateway/modules/crud/mgo/mongo.go:67]: You can set nil before closing the client bro

File: gateway/modules/crud/mgo/mongo.go [hunk 2]
```diff
@@ -119,3 +145,16 @@ func (m *Mongo) GetDBType() model.DBType {
...
+	m.lock.Lock()
+	defer m.lock.Unlock()
+	m.client = c
+}
+
+func (m *Mongo) getClient() *mongo.Client {
+	m.lock.RLock()
+	defer m.lock.RUnlock()
+	client := m.client
+	return client
```
Comment (YourTechBud) [inline gateway/modules/crud/mgo/mongo.go:159]: Club in one line

File: gateway/modules/crud/setters.go [hunk 3]
```diff
@@ -55,7 +55,7 @@ func (m *Module) SetConfig(project string, crud config.DatabaseConfigs) error {
 
 		if m.block != nil {
 			m.block.SetQueryFetchLimit(v.Limit)
-			// Skip if the connection string is the same
+			// Skip if the connection string,dbName & driver config is same
```
Comment (YourTechBud) [inline gateway/modules/crud/setters.go:58]:
```suggestion
			// Skip if the connection string, dbName & driver config is same
```

File: gateway/modules/crud/sql/sql.go [hunk 4]
```diff
@@ -64,12 +87,10 @@ func (s *SQL) IsSame(conn, dbName string, driverConf config.DriverConfig) bool {
...
 // Close gracefully the SQL client
 func (s *SQL) Close() error {
-	if s.client != nil {
-		if err := s.client.Close(); err != nil {
-			return err
-		}
-
-		s.client = nil
+	if s.getClient() != nil {
+		s.setClient(nil)
```
Comment (YourTechBud) [inline gateway/modules/crud/sql/sql.go:91]: Can set nil before closing

**Review by YourTechBud — DISMISSED** [review]

File: gateway/modules/crud/mgo/mongo.go (hunk 2 above)
Comment (sharadregoti) [reply gateway/modules/crud/mgo/mongo.go:159]: HOW

**Review by YourTechBud — APPROVED** [review]

**Pull request merged**